Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/CqlTranslator.java

```diff
@@ -470,9 +521,42 @@ public CqlErrorListener(LibraryBuilder builder, boolean detailedErrors) {
             this.detailedErrors = detailedErrors;
         }
 
+        private VersionedIdentifier extractLibraryIdentifier(cqlParser parser) {
+            RuleContext context = parser.getContext();
+            while (!(context instanceof cqlParser.LibraryContext)) {
+                context = context.parent;
+                if (context == null) {
+                    break;
+                }
+            }
+
+            if (context instanceof cqlParser.LibraryContext) {
```

Comment:
Due to the while loop above, at this point `context` is either `null` or it is a LibraryContext, so this could just be `if (context != null)`.  Again, maybe not necessary...  but I don't remember how Java handles `null instanceof`.

Reply:
While this is true, I prefer to be explicit about the fact that I expect context to be a LibraryContext.